dashboard/app/models/levels/applab.rb

```diff
@@ -125,12 +129,25 @@ def update_json_fields
   end
 
   def validate_maker_if_needed
-    maker_enabled = properties['makerlab_enabled'] == 'true'
+    maker_api_property = properties['makerlab_enabled']
+    maker_enabled = SharedConstants::MAKER_PALETTE_CATEGORIES.keys.include?(maker_api_property)
     starting_category = properties['palette_category_at_start']
-    if Applab.maker_palette_categories.include?(starting_category) && !maker_enabled
+
+    # If the "Maker" palette category is selected, either Maker API must be enabled.
+    if starting_category == 'Maker' && !maker_enabled
+      raise ArgumentError.new(
+        "Selected '#{starting_category}' as the palette category at start, " \
+            "but this level does not have either the circuitPlayground or micro:bit Maker API enabled."
+      )
+    end
+
+    # If a board-specific maker palette ('Circuit' or 'micro:bit') is selected and the corresponding
+    # maker API must also be selected
+    if ['Circuit', 'micro:bit'].include?(starting_category) &&
+      maker_api_property != SharedConstants::MAKER_PALETTE_CATEGORIES.key(starting_category)
```

Comment:
Seems a little odd to get the key from the value, maybe the mapping is backwards here? Looks like this is the only place we do a lookup, so maybe reverse? 

Tangentially, if I understand correctly, it looks like the key in `'maker' => "Maker"` isn't ever used, I wonder if we should remove from the hash and add add "Maker" in `self.maker_palette_categories`

Reply:
This was just a really messy way for me to consolidate all the palette categories into one place so I could reuse it in `self.maker_palette_categories` and minimize the number of times we have strings representing these values. In this case, I grabbed all the values of this to populate the Palette Categories.

However, I realized that the other palette categories were ids and not 'as-they-appear-to-students', so I did some digging and realized that I can do the same - use ids here and have 'friendly' names displayed elsewhere. I went ahead and made those changes, which also beautified the boolean. Hopefully, that all looks good now, but let me know if these concerns remain.

Reply:
Sounds like a great simplification! Sorry for suggesting the mapping idea in the first place, probably was overkill.